src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -874,6 +876,48 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		return false;
 	}
 
+	/**
+	 * Refresh Azure access token if it's expired.
+	 * @param uri connection uri
+	 * @returns true if no need to refresh or successfully refreshed token
+	 */
+	public async refreshAzureAccountTokenIfNecessary(uri: string): Promise<boolean> {
+		let profile = this._connectionStatusManager.getConnectionProfile(uri);
+		if (!profile) {
+			this._logService.info(`Connection not found for uri ${uri}`);
+			return false;
+		}
+
+		//wait for the pending reconnction promise if any
+		let previousReconnectPromise = this._uriToReconnectPromiseMap[uri];
+		if (previousReconnectPromise) {
+			this._logService.info(`Found pending reconnect promise for uri ${uri}, waiting.`);
+			await previousReconnectPromise;
+			delete this._uriToReconnectPromiseMap[uri];
```

Comment:
It looks like you're only deleting this if this function is called again for the same URI. Should clearing the map be done below after the call to .connect?

Reply:
updated